etc/afpd/afprun.c

```diff
@@ -24,20 +24,20 @@
 #include "config.h"
 #endif /* HAVE_CONFIG_H */
 
+#include <errno.h>
+#include <grp.h>
 #include <stdio.h>
 #include <stdlib.h>
+#include <string.h>
+#include <sys/param.h>
 #include <sys/types.h>
-/* #define __USE_GNU 1 */
+#include <sys/wait.h>
 #include <unistd.h>
-#include <grp.h>
 
-#include <errno.h>
-#include <sys/wait.h>
-#include <sys/param.h>
-#include <string.h>
+/* #define __USE_GNU 1 */
 
 /* FIXME */
```

Comment:
Do you know what this FIXME refers to?

Reply:
I have no idea, it's always been there. From the linux man pages:

setresuid() sets the real user ID, the effective user ID, and the saved set-user-ID of the calling process.
seteuid() sets the effective user ID of the calling process. Unprivileged processes may only set the effective user ID to the real user ID, the effective user ID or the saved set-user-ID.

Hmm...

Reply:
Surely something to do with the safety of doing this. Looking at git blame, this fixme goes back to 2005 when the afprun feature was added initially. Something for another changeset, so never mind!